Items Editing/Perfect Timing! - Audio Quantizer.lua

```diff
@@ -0,0 +1,3319 @@
+-- @description Perfect Timing! - Audio Quantizer
+-- @author 80icio
+-- @version 0.21 (Beta)
+-- @screenshot Forum Link https://forum.cockos.com/showthread.php?p=2764162#post2764162
+-- @about
+--   Perfect Timing! is a straightforward audio quantizer with 2 modes, SPLIT and WARP
+--   It features:
+--
+--   - Fast transient analysis and 2 visualization options.
+--   - Timesaving Grid Based transient detection.
+--   - Multitrack and Multichannel analysis and group editing.
+--   - Hipass and Lowpass filters and transient attack for enhanced transient searching.
+--   - Fast quantizing + crossfading.
+--   - Sensitivity and threshold Histogram.
+--   - Low CPU consumption
+--   - Multisized GUI
+--   - Works with fixed lanes
+
+
+--- This script is a re adaptation and re writing of  @Cool Mk Slicer 2 (80icio Mod)
+--- Thanks to Cool for Mk Slicer, I mainly learnt coding LUA from his script
+--- Thanks to amagalma 'Toggle show editing guide line on item under mouse cursor in Main Window' for letting me seee how to draw lines on the screen
+--- Thanks to Nickstomp for export track group names
+--- Thanks to _Stevie_ for testing at very early stages
+--- Thanks to drummerboy, mtierney, MykRobinson for testing
+
+-----------------------------vars vars vars!-----------------------------------
+-------------------------------------------------------------------------------
+-------------------------------------------------------------------------------
+local r = reaper
+local scriptname = 'Perfect Timing! - Audio Quantizer'
+
+function msg (message, clear)
+  if clear then r.ClearConsole() end
+  r.ShowConsoleMsg(tostring(message)..'\n')
+end
+
+
+------------------------------check estensions---------------------------------
+
+function checkIMGUI()
+  local imgui_version, imgui_version_num, REAIMGUI_VERSION = r.ImGui_GetVersion()
+  local version_sum = 0
+  REAIMGUI_VERSION = string.gsub(REAIMGUI_VERSION, '%p', "")
+  return REAIMGUI_VERSION
+end
+
+
+if not dofile(r.GetResourcePath() .. '/Scripts/ReaTeam Extensions/API/imgui.lua') then ------------IMGUI check-----------------------
+  r.ShowMessageBox("Please, install IMGUI extension from REAPACK.\nThanks!", "Ooops!", 0)
+  r.ReaPack_BrowsePackages('ReaImGui: ReaScript binding for Dear ImGui')
+  return false
+else
+  REAIMGUI_VERSION = tonumber(checkIMGUI())
+  if REAIMGUI_VERSION < 83  then ------------ IMGUI version check-----
```

Comment:
This errors out if ReaImGui is not installed (missing imgui.lua), does not properly check its version, and does not activate backward-compatibility.

Correct usage is as follow:

```lua
local imgui_path = r.GetResourcePath() .. '/Scripts/ReaTeam Extensions/API/imgui.lua'
if not r.file_exists(imgui_path) then
  -- ...
  return
end

dofile(imgui_path)('0.8.3')
```

Reply:
Ok thanks! changing it and making a new pull request